### PR TITLE
Bug 1873043: Revise test annotation to reuse rules+code from openshift/kubernetes

### PR DESCRIPTION
**Pull request by marun**
The test annotation code and rules in this repo have been updated to rely on code and rules vendored from openshift/kubernetes. The rules defined in this repo are intended to apply to openshift-only tests and platforms, since rules about kube e2e are inherited from openshift/kubernetes.

Responsibility for maintaining the test annotation code and kube e2e rules was moved to openshift/kubernetes to allow for an openshift-compatible kube e2e job to gate openshift/kubernetes PRs. Such a job can ensure that a PR that includes test changes will be validated against those changes in advance of their incorporation into an openshift-tests binary built from origin.

/cc @sttts @mfojtik @smarterclayton @deads2k

### CODE REVIEW

**Comment by marun** [comment]
I'm abusing the 1.19.0 bump bz because this PR really should have landed at the same time as the 1.19.0-rc.2 PR. Having it remain unmerged has meant lots of manual work to synchronize changes to kube e2e updates to rules.go back to openshift/kubernetes. The origin bump to 1.19.0 will pick those changes back up.

File: test/extended/util/annotate/rules.go [hunk 1]
```diff
@@ -13,180 +20,40 @@ var (
...
-			// TODO(node): configure the cri handler for the runtime class to make this work
-			"should run a Pod requesting a RuntimeClass with a configured handler",
-			"should reject a Pod requesting a RuntimeClass with conflicting node selector",
-			"should run a Pod requesting a RuntimeClass with scheduling",
-
-			// A fix is in progress: https://github.com/openshift/origin/pull/24709
-			`Multi-AZ Clusters should spread the pods of a replication controller across zones`,
-
-			// Disabled as per networking team. Follow-up tracked via https://bugzilla.redhat.com/show_bug.cgi?id=1861214
-			`EndpointSliceMirroring should mirror a custom Endpoints resource through create update and delete`,
```
Comment (sttts) [inline test/extended/util/annotate/rules.go:101]: you went through all of these and made sure that only match kube tests?

Comment (marun) [reply test/extended/util/annotate/rules.go:101]: The reverse. The rules in this file only match origin tests. As per the comment at the top of the file, the openshift/kubernetes equivalent only matches kube tests.

**Comment by sttts** [comment]
/lgtm

README update.

**Pull request merged**